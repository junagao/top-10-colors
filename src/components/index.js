export { default as ColorList } from 'components/ColorList'
export { default as ColorItem } from 'components/ColorItem'
export { default as Header } from 'components/Header'
export { default as Rating } from 'components/Rating'
export { default as Star } from 'components/Star'
