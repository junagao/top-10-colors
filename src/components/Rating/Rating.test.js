import React from 'react'
import { shallow } from 'enzyme'
import { Rating, Star } from 'components'

describe('Rating', () => {
  let wrapper

  const colorMock = {
    id: 'fc352593-c91d-4cf6-b689-7478ee2b61c8',
    name: 'orange',
    hex: '#FFA500',
    rgb: 'rgb(255, 165, 0)',
    rating: 3,
  }

  beforeEach(() => {
    wrapper = shallow(
      <Rating
        id={colorMock.id}
        rating={colorMock.rating}
        totalStars={10}
        onRate={jest.fn()}
      />,
    )
  })

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render one RatingContainer element', () => {
    expect(wrapper.find('RatingContainer').length).toEqual(1)
  })

  it('should render ten instances of Star component', () => {
    expect(wrapper.find(Star).length).toEqual(10)
  })
})
