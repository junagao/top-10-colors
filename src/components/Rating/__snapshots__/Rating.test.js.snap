// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render correctly given the required props 1`] = `
<RatingContainer>
  <Star
    full={true}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={0}
    key="c8501667-9a27-42e8-b3f4-b57b956a8b4b"
    onRate={[MockFunction]}
  />
  <Star
    full={true}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={1}
    key="3b7f0fed-7a58-4f11-a747-200d37f12251"
    onRate={[MockFunction]}
  />
  <Star
    full={true}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={2}
    key="34f4ae5b-4590-45f2-be63-7c6837597d83"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={3}
    key="96a7c411-45e6-4cb0-9638-eb8a2e3e191b"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={4}
    key="c2834dd7-3b51-40d0-8bc2-c0677f3ca4f5"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={5}
    key="98268628-99c7-4b13-b2c3-9310c6914028"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={6}
    key="31c6d110-a9c3-4d94-b44d-2003dc2d2ca3"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={7}
    key="92b93fca-903c-40ef-9227-b052bfbffb03"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={8}
    key="102b3f1c-6b51-4ca7-a916-78cb754b7780"
    onRate={[MockFunction]}
  />
  <Star
    full={false}
    id="fc352593-c91d-4cf6-b689-7478ee2b61c8"
    index={9}
    key="7d3a338c-61bd-4c35-993f-efb451f5713a"
    onRate={[MockFunction]}
  />
</RatingContainer>
`;
